internal/kafka/internal/services/kafkatlscertmgmt/secure_storage.go

```diff
@@ -48,18 +51,41 @@ func newSecureStorage(config *config.AWSConfig, automaticCertificateManagementCo
 		return nil, err
 	}
 	return &secureStorage{
-		secretClient: secretClient,
-		secretCache:  secretCache,
-		secretPrefix: config.SecretManager.SecretPrefix,
+		secretClient:     secretClient,
+		secretCache:      secretCache,
+		lock:             sync.Mutex{},
+		storageItemLocks: map[string]*sync.Mutex{},
+		secretPrefix:     config.SecretManager.SecretPrefix,
 	}, nil
 }
 
+// Lock blocks until it can acquire the name-specific lock.
+// The process of obtaining a name-specific lock is atomic, as required per the certmagic.Locker
+// interface documentation: https://github.com/caddyserver/certmagic/blob/master/storage.go#L87.
```

Comment:
Instead of referring to github code let's refer to the interface documentation: https://pkg.go.dev/github.com/caddyserver/certmagic#Locker

Reply:
thanks.